src/drivers/imx/sai.c

```diff
@@ -33,39 +33,62 @@ static void sai_start(struct dai *dai, int direction)
 
 	uint32_t xcsr = 0U;
```

Comment:
@ranj063 @plbossart I wonder if you can share your experience with IPC here. In stress tests, we get an IPC timeouts for stream STOP command.

Reply:
@dbaluta if you refer to "ipc timed out", yes, you're right. I don't know exactly the root cause. The 3rd commit is optional, but the additional checks don't hurt :)